src/parser.y

```diff
@@ -635,7 +635,7 @@ bool parse(FileModule *&module, const char *text, const std::string &filename, i
   lexerin = NULL;
   parser_error_pos = -1;
   parser_input_buffer = text;
-  parser_sourcefile = path;
+  parser_sourcefile = std::make_shared<fs::path>(fs::absolute(fs::path(filename)));
```

Comment:
We already have a local variable `path`, which could be passed directly to `make_shared()`